storm-webapp/src/test/java/org/apache/storm/daemon/ui/UIHelpersTest.java

```diff
@@ -0,0 +1,543 @@
+        topologyStats.set_window_to_acked(new HashMap<>());
+        topologyStats.set_window_to_complete_latencies_ms(new HashMap<>());
+        topologyStats.set_window_to_failed(new HashMap<>());
+
+        // Create empty AggregateStats instances.
+        final Map<String,ComponentAggregateStats> idToSpoutAggStats = new HashMap<>();
+
+        final Map<String,ComponentAggregateStats> idToBoltAggStats = new HashMap<>();
+
+        // Build up TopologyPageInfo instance
+        topoPageInfo = new TopologyPageInfo();
+        topoPageInfo.set_topology_conf(topoConfigJson);
+        topoPageInfo.set_id(TOPOLOGY_ID);
+        topoPageInfo.set_topology_stats(topologyStats);
+        topoPageInfo.set_id_to_spout_agg_stats(idToSpoutAggStats);
+        topoPageInfo.set_id_to_bolt_agg_stats(idToBoltAggStats);
+    }
+
+    /**
+     * Very narrow test case to validate that 'last error' fields are populated for a bolt
+     * with an error is present.
+     */
+    @Test
+    void test_getTopologyBoltAggStatsMap_includesLastError() {
+        // Define inputs
+        final String expectedBoltId = "MyBoltId";
+        final String expectedErrorMsg = "This is my test error message";
+        final int expectedErrorTime = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+        final int expectedErrorPort = 4321;
+        final String expectedErrorHost = "my.errored.host";
+
+        // Define our Last Error
+        final ErrorInfo expectedLastError = new ErrorInfo(expectedErrorMsg, expectedErrorTime);
+        expectedLastError.set_port(expectedErrorPort);
+        expectedLastError.set_host(expectedErrorHost);
+
+        // Build stats instance for our bolt
+        final ComponentAggregateStats aggregateStats = buildBoltAggregateStatsBase();
+        aggregateStats.set_last_error(expectedLastError);
+        addBoltStats(expectedBoltId, aggregateStats);
+
+        // Call method under test.
+        final Map<String, Object> result = UIHelpers.getTopologySummary(
+            topoPageInfo,
+            WINDOW,
+            new HashMap<>(),
+            "spp"
+        );
+
+        // Validate
+        assertNotNull(result, "Should never return null");
+
+        // Validate our Bolt result
+        final Map<String, Object> boltResult = getBoltStatsFromTopologySummaryResult(result, expectedBoltId);
+        assertNotNull(boltResult, "Should have an entry for bolt");
+
+        // Verify each piece
+        assertEquals(expectedBoltId, boltResult.get("boltId"));
+        assertEquals(expectedBoltId, boltResult.get("encodedBoltId"));
+
+        // Verify error
+        assertEquals(expectedErrorMsg, boltResult.get("error"));
+        assertEquals(expectedErrorMsg, boltResult.get("lastError"));
+        assertEquals(expectedErrorPort, boltResult.get("errorPort"));
+        assertEquals(expectedErrorHost, boltResult.get("errorHost"));
+        assertEquals(expectedErrorTime, boltResult.get("errorTime"));
+
+        // Fuzzy matching
```

Comment:
You can use simulatedTime https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/utils/Time.java#L30. 

And then the check can be accurate.